Kernel/Unit.hpp

```diff
@@ -168,6 +169,35 @@ class Unit
   static unsigned _firstNonPreprocessingNumber;
 
   static unsigned _lastParsingNumber;
+
+private:
+  template<class Pre, class Post>
+  void _traverseParents(Unit& self, unsigned& depth, Pre& pre, Post& post) {
+    pre(unsigned(depth), &self);
+    depth++;
+    auto infit = self.inference().iterator();
+    while (self.inference().hasNext(infit)) {
+      auto parent = self.inference().next(infit);
+      _traverseParents(*parent, depth, pre, post);
```

Comment:
Yes indeed. It's only supposed to be used for debugging (i.e. outputting to be inspected by a human) and when the output is so long that it will overflow the stack it will surely overflow the inspector's brain as well :)